src/ChessBoard/MovePieces.tsx

```diff
@@ -1,134 +1,493 @@
 
-    componentDidMount() {
-        const canvas = this.canvasRef.current!;
+    useEffect(() => {
+        const canvas = canvasRef.current!;
         const ctx = canvas.getContext("2d")!;
-        window.addEventListener("mousemove", this.handleMouseMove);
-        canvas.addEventListener("mousemove", this.onMouseMove);
-        canvas.addEventListener("mouseup", this.onMouseUp);
-        this.draw_pieces(ctx);
-    }
-
-    componentWillUnmount() {
-        const canvas = this.canvasRef.current!;
-        canvas.removeEventListener("mousemove", this.onMouseMove);
-        canvas.removeEventListener("mouseup", this.onMouseUp);
-        window.removeEventListener("mousemove", this.handleMouseMove)
-    }
+        window.addEventListener("mousemove", handleMouseMove);
+        canvas.addEventListener("mousemove", onMouseMove);
+        canvas.addEventListener("mouseup", onMouseUp);
+        draw_pieces(ctx);
+        
+        return (() => {
+            canvas.removeEventListener("mousemove", onMouseMove);
+            canvas.removeEventListener("mouseup", onMouseUp);
+            window.removeEventListener("mousemove", handleMouseMove)
+        })
+    }, [canvasRef])
 
-    getMousePositions(event: any){
-        const canvas = this.canvasRef.current!;
+    function getMousePositions(event: any){
+        const canvas = canvasRef.current!;
         const { left, top } = canvas.getBoundingClientRect();
         return {
             x: event.clientX - left,
             y: event.clientY - top,
-        };
+        }
     }
 
-    onMouseDown = (event: React.MouseEvent<HTMLCanvasElement>) => {
-        const mousePositions = this.getMousePositions(event)
-        const index = this.getIndexAtPosition(mousePositions.x, mousePositions.y);
+    function onMouseDown (event: React.MouseEvent<HTMLCanvasElement>) {
+        const mousePositions = getMousePositions(event)
+        const index = getIndexAtPosition(mousePositions.x, mousePositions.y);
         if (index !== -1) {
-            this.setState({ isDragging: true, draggingIndex: index });
-            window.addEventListener("pointermove", this.onMouseMove);
-            window.addEventListener("pointerup", this.onMouseUp);
-            this.setState({ mousePosition: { x: mousePositions.x, y: mousePositions.y } });
+            draggingIndex = index
+            mousePosition = {x: mousePositions.x, y: mousePositions.y};
         }
     }
 
-    onMouseUp = () => {
-        this.setState({ isDragging: false, draggingIndex: -1 });
-        window.removeEventListener("pointermove", this.onMouseMove);
-        window.removeEventListener("pointerup", this.onMouseUp);
+    const onMouseUp = () => {
+        adjustPiecePosition()
+        draggingIndex = -1
     }
 
-    handleMouseMove = (event: MouseEvent) => {
-        const mousePositions = this.getMousePositions(event)
+    const handleMouseMove = (event: MouseEvent) => {
+        const mousePositions = getMousePositions(event)
```

Comment:
You can make it shorter and wihout duplicating variables names with destrucurization. Example: `let {x, y} = getMousePositions(event)`